include/linker/devicetree_sram.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT mmio_sram
+
+#define _DT_SRAM_INST_NAME(inst)	DT_STRING_TOKEN(DT_DRV_INST(inst), label)
```

Comment:
The use of the `label` property for region naming was already discussed and rejected in #36365